### ISSUE TITLE
Incorrect source position before bogus comment

### ISSUE THREAD

**Issue by labdsf** [ndmitchell/tagsoup#71]
Bug #70 is fixed, but the same bug with bogus comments is not:
```
> parseTagsOptions Text.HTML.TagSoup.parseOptions{ optTagWarning = True, optTagPosition = True } "<div><!--foo-->bar</div>"
[TagPosition 1 1,TagOpen "div" [],TagPosition 1 6,TagComment "foo",TagPosition 1 16,TagText "bar",TagPosition 1 19,TagClose "div"]
> parseTagsOptions Text.HTML.TagSoup.parseOptions{ optTagWarning = True, optTagPosition = True } "<div><?foo</div>"
[TagPosition 1 1,TagOpen "div" [],TagPosition 1 8,TagOpen "?foo<" [("div","")],TagPosition 1 13,TagWarning "Unexpected \"/\"",TagPosition 1 17,TagWarning "Expected \"?>\""]
```

Note the `TagPosition 1 8` in the second example.

**Comment by ndmitchell** [comment]
The second one now gives:

```
[TagPosition 1 1,TagOpen "div" [],TagPosition 1 6,TagOpen "?foo<" [("div","")],TagPosition 1 13,TagWarning "Unexpected \"/\"",TagPosition 1 17,TagWarning "Expected \"?>\""]
```

Which is what I believe you were hoping for.

**Comment by ndmitchell** [comment]
I've released 0.14.4 with the fix included. Let me know if you find any more issues.
